sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -453,6 +453,12 @@ final class DataFrameWriter private[sql](df: DataFrame) {
    *   format("json").save(path)
    * }}}
    *
+   * You can set the following JSON-specific options for writing JSON files:
+   * <li>`compression` or `codec` (default `null`): compression codec to use when saving to file.
+   * This should be the fully qualified name of a class implementing
+   * [[org.apache.hadoop.io.compress.CompressionCodec]] or one of the known case-insensitive
```

Comment:
I'd remove the mention of org.apache.hadoop.io.compress.CompressionCodec

The hadoop API is so difficult to use that I don't think it deserves to be user-facing here.